tools/test-framework/src/chain/driver.rs

```diff
@@ -164,7 +173,11 @@ impl ChainDriver {
 
     /// Returns the full URL for the GRPC address.
     pub fn grpc_address(&self) -> String {
-        format!("http://localhost:{}", self.grpc_port)
+        if self.ipv6_grpc {
+            format!("http://[::1]:{}", self.grpc_port)
+        } else {
+            format!("http://localhost:{}", self.grpc_port)
```

Comment:
To make it clearer that this uses IPv4:

```suggestion
            format!("http://127.0.0.1:{}", self.grpc_port)
```